src/coreclr/dlls/mscordac/mscordac_unixexports.src

```diff
@@ -6,8 +6,8 @@ CLRDataCreateInstance
 
 ; PAL module registration
 DllMain
-PAL_RegisterModule
-PAL_UnregisterModule
+#PAL_RegisterModule
```

Comment:
This won't work because the the PAL loader code [here](https://github.com/dotnet/runtime/blob/efe93b7d30add5ae40a57b92140990a522e7ee9c/src/coreclr/pal/src/loader/module.cpp#L1709) and [here](https://github.com/dotnet/runtime/blob/efe93b7d30add5ae40a57b92140990a522e7ee9c/src/coreclr/pal/src/loader/module.cpp#L1266) expects the name to be undecorated.  This PAL loader code can't really be changed because the diagnostics repo PAL also does it.

Reply:
Ah, just to be clear; currently this is the situation:

Before (`main` branch):
```sh
$ nm -D .dotnet/shared/Microsoft.NETCore.App/8.0.0-alpha.1.23057.5/libmscordaccore.so | grep -iE 'PAL_.*eg'
0000000000206040 T PAL_RegisterModule
0000000000206260 T PAL_UnregisterModule

$ nm -D .dotnet/shared/Microsoft.NETCore.App/8.0.0-alpha.1.23057.5/libmscordbi.so | grep -iE 'PAL_.*eg'
# nothing
```

After (PR branch):
```sh
$ nm -D artifacts/bin/coreclr/linux.x64.Debug/libmscordaccore.so | grep -iE 'PAL_.*eg'
000000000014119c T DAC_PAL_RegisterModule
00000000001411a1 T DAC_PAL_UnregisterModule

$ nm -D artifacts/bin/coreclr/linux.x64.Debug/libmscordbi.so | grep -iE 'PAL_.*eg'
                 U DAC_PAL_RegisterModule
                 U DAC_PAL_UnregisterModule
00000000000d293c T PAL_RegisterModule
00000000000d2941 T PAL_UnregisterModule
```

and we want `libmscordaccore.so` to have un-decorated `PAL_{Unr,R}egisterModule`, right? I'll try your patch.

Reply:
Correct.